### ISSUE TITLE
Issues v1

### ISSUE THREAD

**Issue by sharky127** [andreaslarsen/YosemiForge#1]
i cloned your rep on sourcetree. whats next?

**Comment by andreaslarsen** [comment]
open FontForge and make sure you're using the Windyforge.theme located in your local clone.

**Comment by sharky127** [comment]
done

**Comment by andreaslarsen** [comment]
let's try and fix the buttons first

**Comment by sharky127** [comment]
just so you know, i havo 0 experience with github/sourcetree

**Comment by andreaslarsen** [comment]
In fontforge: chose file, x resources editor (in the bottom)

**Comment by andreaslarsen** [comment]
I figured that out already, no problem

![screen shot 2015-04-20 at 21 52 10](https://cloud.githubusercontent.com/assets/3859425/7239108/9319c716-e7a7-11e4-8296-cdad1b376253.png)

**Comment by andreaslarsen** [comment]
How does the Gadget window look:
![screen shot 2015-04-20 at 21 53 51](https://cloud.githubusercontent.com/assets/3859425/7239132/c34459ec-e7a7-11e4-8bfb-36ca31b07275.png)

**Comment by sharky127** [comment]
yup

**Comment by andreaslarsen** [comment]
And button like this?

![screen shot 2015-04-20 at 21 56 10](https://cloud.githubusercontent.com/assets/3859425/7239177/151a846c-e7a8-11e4-98ee-6ab324b8ec2d.png)

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7239204/a9e5e2a0-e7b0-11e4-89c0-e726d38c8c08.PNG)

**Comment by andreaslarsen** [comment]
Try fixing the top part so it looks the same as mine and then save as...

**Comment by sharky127** [comment]
do i save it on the source .theme file, or make a copy ?

**Comment by andreaslarsen** [comment]
on the source

**Comment by sharky127** [comment]
done

**Comment by andreaslarsen** [comment]
the quit fontforge and start again

**Comment by andreaslarsen** [comment]
the default button should hopefully look better now, no?

**Comment by andreaslarsen** [comment]
the ok and cancel button might have some ugly gradients (this is a fontforge bug that we fix in the end)

**Comment by sharky127** [comment]
which buttons are you looking at?

**Comment by andreaslarsen** [comment]
e.g. the "save as..." button from before?

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7239326/c507575c-e7b1-11e4-8555-b3754c6778c0.PNG)

**Comment by sharky127** [comment]
seems the same

**Comment by andreaslarsen** [comment]
yep, one step forward and one step back. I will now post a series of screenshots and you can just try to copy the settings so they are the same and then "save as..." every once in a while (if your fontforge has a tendency to crash)

**Comment by sharky127** [comment]
got it

**Comment by andreaslarsen** [comment]
In the mean time.
You have now made some changes to your local clone

**Comment by andreaslarsen** [comment]
You will now commit those changes so go back to sourcetree app

**Comment by andreaslarsen** [comment]
switch to file status view

**Comment by andreaslarsen** [comment]
view -> file status view

**Comment by sharky127** [comment]
yup

**Comment by andreaslarsen** [comment]
Tick the "unstated changes" box

**Comment by sharky127** [comment]
hmm

**Comment by andreaslarsen** [comment]
*unstaged, sry

**Comment by sharky127** [comment]
dont see it, do i have to open something on the main window ?
 like it says "nothing to commit"

**Comment by andreaslarsen** [comment]
Screenshot from another project 
![screen shot 2015-04-20 at 22 12 07](https://cloud.githubusercontent.com/assets/3859425/7239457/51dfa6aa-e7aa-11e4-90cc-dba7c10d4cc1.png)

**Comment by andreaslarsen** [comment]
you don't have WindyForge.theme in the "Unstaged files" list?

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7239492/f4a32bf2-e7b2-11e4-9dc7-4c5e491bcb0e.PNG)

**Comment by andreaslarsen** [comment]
and you are sure you saved the changes directly to the cloned file?
Try renaming one of the files in the cloned folder and see if it pops up

**Comment by sharky127** [comment]
i saved the changes to the .theme file which i was using (not in the cloned folder)

**Comment by sharky127** [comment]
should the ResourceFIle in preferences be directed to the .theme file in the clone folder (sourcetree) ?

**Comment by andreaslarsen** [comment]
yep, sorry if that wasn't clear

**Comment by sharky127** [comment]
ok. doesnt let me check unstaged files

**Comment by sharky127** [comment]
cant uncheck Staged Files

**Comment by sharky127** [comment]
checks itself right back

**Comment by andreaslarsen** [comment]
try typing the commit message first 

e.g. "test"

**Comment by sharky127** [comment]
git -c diff.mnemonicprefix=false -c core.quotepath=false commit -q -F C:\Users\gauro_000\AppData\Local\Temp\sbd2c0db.exd

*** Please tell me who you are.

Run


  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"





to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'gauro_000@Bob.(none)')







Completed with errors, see above.

**Comment by andreaslarsen** [comment]
![screen shot 2015-04-20 at 22 26 29](https://cloud.githubusercontent.com/assets/3859425/7239719/529af534-e7ac-11e4-999d-6a10413bdb88.png)

**Comment by andreaslarsen** [comment]
ah, you need to be signed in

**Comment by sharky127** [comment]
alright

**Comment by sharky127** [comment]
i signed in and im back to "nothing to commit"

**Comment by andreaslarsen** [comment]
ok, now try to rename a file

**Comment by andreaslarsen** [comment]
inside the cloned folder

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7239763/2af919f8-e7b5-11e4-8783-78dd038bbce7.PNG)

**Comment by andreaslarsen** [comment]
perfect, commit that

**Comment by sharky127** [comment]
when i press commit, it goes back to the "nothing to commit" window i linked above

**Comment by sharky127** [comment]
"nothing to commit" with an "open in explorer" button

**Comment by andreaslarsen** [comment]
sry, tick the small "Push changes imme...."

**Comment by andreaslarsen** [comment]
or

**Comment by andreaslarsen** [comment]
notice the push icon has a red "1"

**Comment by sharky127** [comment]
yup

**Comment by andreaslarsen** [comment]
try clicking on it

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7239848/ac5e43b0-e7b5-11e4-9e4c-04f9786ac872.PNG)

**Comment by andreaslarsen** [comment]
yes

**Comment by sharky127** [comment]
"ok" ?

**Comment by andreaslarsen** [comment]
yep

**Comment by sharky127** [comment]
pushing

**Comment by andreaslarsen** [comment]
don't be afraid to try things out - with git comes full versioning so you can always take a step backwards

**Comment by sharky127** [comment]
well i have no clue what im doing, so i think i should do exactly what you tell me to do for best results

**Comment by sharky127** [comment]
still pushing

**Comment by andreaslarsen** [comment]
I don't know why it's that slow - usually takes a second or two

**Comment by sharky127** [comment]
done

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7239970/6e932b08-e7b6-11e4-8269-10cf31aaf0f8.PNG)
 im here now

**Comment by andreaslarsen** [comment]
Pushing: When you "push" a change from your local clone to e.g. github
Pulling: When you "pull" a change from Github to your local clone (if for instance I changed a file)
Commits: One push or pull can contain multiple commits

**Comment by andreaslarsen** [comment]
You can see that you've succeeded by looking at commits:

https://github.com/andreaslarsen/YosemiForge/commits/master

**Comment by sharky127** [comment]
so its basically and adjustment to the github files?

**Comment by andreaslarsen** [comment]
yes, this is a very small project but if you imagine a bigger app like e.g. fontforge then this process makes a lot of sense.

What we are working on is actually a fork of the theme that I moved to the fontforge repo
![screen shot 2015-04-20 at 22 43 27](https://cloud.githubusercontent.com/assets/3859425/7240067/b57595fe-e7ae-11e4-9409-8c928d8ba022.png)

**Comment by andreaslarsen** [comment]
if you open sourcetree again

**Comment by andreaslarsen** [comment]
maybe refresh

**Comment by andreaslarsen** [comment]
there should be something for you to pull

**Comment by sharky127** [comment]
should i pull it ?

**Comment by andreaslarsen** [comment]
yes, you can inspect what it is prior to pulling (you should be able to see somewhere in the interface that I made changes to the readme file)

**Comment by andreaslarsen** [comment]
not important now, just so you're aware of it in the future

**Comment by sharky127** [comment]
done

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7240130/8fe5e43e-e7b7-11e4-85b3-f09ad3af23c0.PNG)

**Comment by andreaslarsen** [comment]
and you've basically completed the basics of gits. It's that simple :smile:

**Comment by andreaslarsen** [comment]
now I'll be adding some screenshots and when you copy the settings and save them to your local file then you can commit it to the master, so that I, you or someone else can pick up from where you left.

**Comment by sharky127** [comment]
its still a dark forest for me, ill have to spend sime time getting used to everything.

**Comment by sharky127** [comment]
ok, remember that all the instructions should be eli5

**Comment by andreaslarsen** [comment]
Don't worry, I'm nowhere near a wizard but the trick is to read the error messages carefully and then ignore all the fancy commands it tells you to use. You can use the sourcetree app + github web interface for 99% of all things.

**Comment by andreaslarsen** [comment]
I'll now add a bunch of screenshots of how the "x resource editor" looks in my fontforge. 

Ignore the colors (we will fix those in the end). 
Important thing is to get the buttons and fonts to look right.

**Comment by andreaslarsen** [comment]
so open fontforge again

**Comment by andreaslarsen** [comment]
file -> x resource editor

**Comment by sharky127** [comment]
alright, so im working on fontforge now with the ResourceFile .theme in the clone folder of sourcetree

**Comment by andreaslarsen** [comment]
yes

**Comment by andreaslarsen** [comment]
The way theming works in fontforge is many settings "walk" down...so that e.g. "List Mark" inherits many settings from "GGadget"

**Comment by andreaslarsen** [comment]
So we start at the top: (you don't need to post screenshots) just make sure the settings look the same

![screen shot 2015-04-20 at 22 23 24](https://cloud.githubusercontent.com/assets/3859425/7240308/65dd1a6a-e7b0-11e4-8dc8-1622eb9c90e5.png)

**Comment by andreaslarsen** [comment]
Again: ignore the colors, just check that font size and font family are the same as my screenshot

**Comment by andreaslarsen** [comment]
![screen shot 2015-04-20 at 22 58 13](https://cloud.githubusercontent.com/assets/3859425/7240373/c0cf62f2-e7b0-11e4-8857-e1fe9415e6e3.png)

**Comment by andreaslarsen** [comment]
Nothing in "outline view 2"

**Comment by andreaslarsen** [comment]
![screen shot 2015-04-20 at 23 00 43](https://cloud.githubusercontent.com/assets/3859425/7240419/176e94d4-e7b1-11e4-87dc-163fe60e1dc2.png)

**Comment by andreaslarsen** [comment]
![screen shot 2015-04-20 at 23 01 25](https://cloud.githubusercontent.com/assets/3859425/7240433/30cbcfb4-e7b1-11e4-809d-4b41018d9080.png)

**Comment by andreaslarsen** [comment]
![screen shot 2015-04-20 at 23 01 57](https://cloud.githubusercontent.com/assets/3859425/7240448/4835a760-e7b1-11e4-8d80-3067d6f4feb8.png)

**Comment by andreaslarsen** [comment]
![screen shot 2015-04-20 at 23 02 25](https://cloud.githubusercontent.com/assets/3859425/7240463/57bea24a-e7b1-11e4-9755-c0a15c5238b3.png)

**Comment by sharky127** [comment]
done

**Comment by andreaslarsen** [comment]
Great. I'll collect the rest of the screenshot in the next post so it might take a while. But you can always come back. 

A small Git recap: 

* You are now making changes to a local clone of a git that you and I are currently editing.
* When you have made some changes then you can commit them. (after you've "saved as..." in fontforge)
* When we are done those changes will be transferred from this fork that we are working on into the main fontforge repo
* So that the theme you've helped create will be used by many fontforge users in the future and you've contributed directly to open source software development

**Comment by andreaslarsen** [comment]
This is how fontforge and many other apps are being developed. You can see the fontforge app is a little bit bigger than our project (e.g. 17.000 something commits) https://github.com/fontforge/fontforge

**Comment by sharky127** [comment]
Wow, that's pretty cool. Never knew how the open source development worked until now

**Comment by andreaslarsen** [comment]
You can see that our fork is currently 3 commits ahead of the fontforge version and detail view of everything that has changed. https://github.com/fontforge/YosemiForge/compare/master...andreaslarsen:master

**Comment by andreaslarsen** [comment]
sorry, 4 with your latest commit

**Comment by andreaslarsen** [comment]
And now that you're a github expert you can fork any project - maybe a jQuery plugin for a website or a open source font or something else - edit it and make it better and then send the changes back to where you forked from so that everyone can benefit.

**Comment by sharky127** [comment]
Haha, yeah there's some reading and learning i have to do before that

**Comment by andreaslarsen** [comment]
no no, your first commit to another project can be as simple as updating some text in a readme or translating something into your native language or whatever

**Comment by sharky127** [comment]
true, but one would be a lot more valuable with more advanced skillsets other than editing .txt's

**Comment by andreaslarsen** [comment]
One last thing: 
the process of sending back is called "pull requests" - so when we are done we'll ask fontforge team to "pull" the changes we have made by creating that "pull request" and when they accept (and "they" will since I am an admin in that team). 

So some basic git words to familiarise yourself with would be.

* Commits - you have made some of those
* Pull - you must always check if the pull button in sourcetreeapp has a small red number and pull before you push (if there is something to pull)
* Push - you have pushed some commits already

There are some good videos if you search for them and https://guides.github.com is also a good place to start

**Comment by sharky127** [comment]
Thanks. Yeah i'll have to do "some" learning in the future

**Comment by andreaslarsen** [comment]
Jup, I've added all the screenshots in a Yosemiforge/Resources/setting screenshots

**Comment by andreaslarsen** [comment]
so you should be able to pull those from within SourceTree

**Comment by sharky127** [comment]
done. so now what? check all the screencaps ?

**Comment by andreaslarsen** [comment]
yep, sort them by date with the oldest first in file explorer and they should be ordered (top to bottom)

**Comment by andreaslarsen** [comment]
when that is done it should hopefully look better. 

And if you at some point can't understand why sourcetree won't let you push it's probably because I  have updated something and you need to pull that first - it's so easy to overlook that little red icon when you are pushing away.

**Comment by andreaslarsen** [comment]
ups, some of the screenshots may be wrong so hold your horses (sorry)

**Comment by andreaslarsen** [comment]
FontForge x resource editor has a way of adding stuff you didn't ask it to do...

**Comment by sharky127** [comment]
done

**Comment by sharky127** [comment]
wow its horrible now

**Comment by andreaslarsen** [comment]
great, I'm afraid there were some errors in the screenshots (I only look thoroughly on the first ones and went screenshot crazy with the rest.)

**Comment by sharky127** [comment]
![capture](https://cloud.githubusercontent.com/assets/12038051/7241365/2a28d9a8-e7c1-11e4-8531-36e334833517.PNG)

**Comment by andreaslarsen** [comment]
jep, my x resources editor have done all sorts of crazy stuff...I will fix it and also the screenshots but it may take a few minutes...

**Comment by sharky127** [comment]
i cant even see he buttons, since the open font window doesnt scroll

**Comment by andreaslarsen** [comment]
try choosing, file, preferences and then choose the YosemiForge.theme

**Comment by andreaslarsen** [comment]
it should take you back to where you started -> then save as WindyForge.theme

**Comment by sharky127** [comment]
i cant. Open File is the first window i get when i run fontforge

**Comment by andreaslarsen** [comment]
ah, do it manually then.
quit fontforge

**Comment by andreaslarsen** [comment]
copy YosemiForge.theme

**Comment by andreaslarsen** [comment]
paste it in the same folder as WindyForge.theme

**Comment by andreaslarsen** [comment]
delete WindyForge.theme

**Comment by andreaslarsen** [comment]
rename YosemiForge.theme -> WindyForge.theme

**Comment by andreaslarsen** [comment]
reopen fontforge

**Comment by sharky127** [comment]
done

**Comment by andreaslarsen** [comment]
good - commit your changes and I'll fix it from here and be back in the morning with verified screenshots after I've regained control over my "x resources editor"....

**Comment by sharky127** [comment]
i commited the changes which messed up my fontforge client. But now i have a fresh WindyForge.theme in the clone folder bc i had to replace it

**Comment by sharky127** [comment]
good?

**Comment by andreaslarsen** [comment]
good! I'll be back in the morning so just check in tomorrow whenever you have time and we should be able to fix this in a flash :smile: 

I hope you've learned something tonight so that my screenshot screwup hasn't killed your appetite for helping out ;)

**Comment by sharky127** [comment]
Alright, ill check in tomorrow. Thanks for the info!

**Comment by andreaslarsen** [comment]
Everything should be ready for you.

1. Sourcetree
  * Pull my latest commits
  * So your local clone is up to date
2. Open FontForge
  * x resource editor
  * "copy" the screenshots (they are correct now, I promise;)
  * save as... -> Windyforge.theme
3. Quit FontForge
4. Open FontForge
5. How does it look?

Some text strings that might come in handy:
```
400 10pt Clear Sans
```
```
~/.FontForge/tango-ct.theme:~/.FontForge/tango-ct.theme/tango:/Applications/FontForge.app/Contents/Resources/opt/local/share/fontforge/pixmaps/
```

**Comment by andreaslarsen** [comment]
@sharky127 Make sure you're pulling the latest updates I just committed before "copying" the screenshots as I've just made some updates.
